### PR TITLE
Story form

### PR DESCRIPTION
**Pull request by modrayyes**
image

### CODE REVIEW

**Review by NadiaJalalTamimi — APPROVED** [review]

**Review by reemsh2000 — APPROVED** [review]

File: client/src/components/uploadStoryForm/index.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+          placeholder="العنوان"
+        />
+        <>
+          <TextArea
+            rows={4}
+            placeholder="القصة"
+            onChange={(e) => setStory((prev) => ({ ...prev, content: e.target.value }))}
+          />
+          <form onSubmit={handleFileInputChange} className="form">
+            <button type="button" className="buttonS">
```
Comment (Nizar7zak) [inline client/src/components/uploadStoryForm/index.js:88]: we have a common component button, it takes a class name and text as props

File: client/src/components/uploadStoryForm/index.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+              <dev className="inputForm">
+                <label htmlFor="fileInput">تحميل الصورة</label>
+                <input
```
Comment (Nizar7zak) [inline client/src/components/uploadStoryForm/index.js:91]: here you use input html and in line 77 used Input from antd, just be consist

**Review by Nizar7zak — CHANGES_REQUESTED** [review]

File: client/src/components/uploadStoryForm/index.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+import { Modal, Button, Input } from 'antd';
+import './styles.css';
+import httpService from '../../services/httpService';
+// import axios from 'axios';
+
+const { TextArea } = Input;
+
+function StoryForm() {
+  const [fileInputState, setFileInputState] = useState('');
+  const [selectedFile, setSelectedFile] = useState();
```
Comment (LinaYahya) [inline client/src/components/uploadStoryForm/index.js:12]: give it default value

File: client/src/components/uploadStoryForm/index.js [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    reader.onloadend = () => {
+      uploadImage(reader.result);
+    };
+  };
+
+  return (
+    <>
+      <Button type="primary" onClick={() => setVisible(true)}>
+        أضف قصتك
+      </Button>
```
Comment (LinaYahya) [inline client/src/components/uploadStoryForm/index.js:66]: this button will be out of this component, and visible and setVisisble will pass as a props to the component

File: client/src/components/uploadStoryForm/index.js [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+        />
+        <>
+          <TextArea
+            rows={4}
+            placeholder="القصة"
+            onChange={(e) => setStory((prev) => ({ ...prev, content: e.target.value }))}
+          />
+          <form onSubmit={handleFileInputChange} className="form">
+            <button type="button" className="buttonS">
+              <dev className="inputForm">
```
Comment (LinaYahya) [inline client/src/components/uploadStoryForm/index.js:89]:
```suggestion
              <div className="inputForm">
```

File: client/src/components/uploadStoryForm/index.js [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+          onChange={(e) => setStory((prev) => ({ ...prev, title: e.target.value }))}
+          placeholder="العنوان"
+        />
+        <>
+          <TextArea
+            rows={4}
+            placeholder="القصة"
+            onChange={(e) => setStory((prev) => ({ ...prev, content: e.target.value }))}
+          />
+          <form onSubmit={handleFileInputChange} className="form">
```
Comment (LinaYahya) [inline client/src/components/uploadStoryForm/index.js:87]: the form will wrap all our inputs not only file input, although notice that content and title are required in database so those values shouldn't be empty.
As we validate the request body on the server side we need to validate input's here else.
So, If I'm a user trying to submit form with empty values I should see an error message clarify that to me.
https://ant.design/components/form/ check the first example here to add rule to your input

File: client/src/components/uploadStoryForm/index.js [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+        onOk={handleSubmitFile}
+        okText="موافق"
+        cancelText="الغاء"
+        onCancel={() => setVisible(false)}
+        width={500}
+      >
+        <Input
+          onChange={(e) => setStory((prev) => ({ ...prev, title: e.target.value }))}
+          placeholder="العنوان"
+        />
```
Comment (LinaYahya) [inline client/src/components/uploadStoryForm/index.js:80]:
```suggestion
      <Form.Item
        label="اللعنوان"
        name="عنوان"
        rules={[{ required: true, message: 'أضف عنواناً لقصتك' }]}
      >
        <Input
          onChange={(e) => setStory((prev) => ({ ...prev, title: e.target.value }))}
          placeholder="العنوان"
        />
        <Form.Item />
```

File: client/src/components/uploadStoryForm/index.js (hunk 6 above)
Comment (LinaYahya) [reply client/src/components/uploadStoryForm/index.js:87]: form onSubmit here will never work,
Instead make a ref to your form and handle onSubmit modal to run this function

**Review by LinaYahya — CHANGES_REQUESTED** [review]

**Review by Nizar7zak — APPROVED** [review]
Great Job Mohammed

**Pull request merged**